File: benchmark/template/programs/go/adversarial/ADV-19.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func f(n int64) int64 {
	return 1 + f(n+1)
}

func main() {
	fmt.Println("ADV-START")

	sc := bufio.NewScanner(os.Stdin)
	sc.Scan()
	n, _ := strconv.ParseInt(sc.Text(), 10, 64)

	r := f(n)

	fmt.Printf("OBS=R:%d\n", r)
	fmt.Println("ADV-END")
}

File: benchmark/template/programs/zig/adversarial/ADV-19.zig
const std = @import("std");

fn f(n: i64) i64 {
    return 1 + f(n + 1);
}

pub fn main() !void {
    var t = std.Io.Threaded.init_single_threaded;
    const io = t.io();
    const out = std.Io.File.stdout();
    try out.writeStreamingAll(io, "ADV-START\n");

    var ib: [64]u8 = undefined;
    var rd = std.Io.File.stdin().readerStreaming(io, &ib);
    const start: i64 = try std.fmt.parseInt(i64, try rd.interface.takeSentinel('\n'), 10);

    const r: i64 = f(start);

    var ob: [64]u8 = undefined;
    try out.writeStreamingAll(io, try std.fmt.bufPrint(&ob, "OBS=R:{d}\n", .{r}));
    try out.writeStreamingAll(io, "ADV-END\n");
}
